src/test_graph.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding utf-8 -*-
+import pytest
+
+
+TABLE = [(
+    [1, 2, 3],
+    [[2, 3], [], [1]]
+)]
+
+
+@pytest.fixture
+def initial_graph():
+    graph = {1: [2, 3],
+             2: [],
+             3: [1]}
+    return graph
```

Comment:
You only really ever test one type of graph, or an empty one. You should potentially come up with alternative ones to test.